BusinessLogicLayer/Services/CourseService.cs

```diff
@@ -5,17 +5,25 @@
     using BusinessLogicLayer.Interfaces;
     using DataAccessLayer.Entities;
     using DataAccessLayer.Interfaces;
+    using DataAccessLayer.Repositories;
+    using EducationPortal.BLL.Interfaces;
+    using EducationPortal.Domain.Comparers;
     using Entities;
 
     public class CourseService : ICourseService
     {
         private readonly IRepository<Course> courseRepository;
         private readonly ISkillService skillService;
+        private ICourseComparerService courseComparer;
 
-        public CourseService(IRepository<Course> repository, ISkillService skillService)
+        public CourseService(
+            IEnumerable<IRepository<Course>> repositories,
+            ISkillService skillService,
+            ICourseComparerService courseComparer)
         {
-            this.courseRepository = repository;
+            this.courseRepository = repositories.FirstOrDefault(t => t.GetType() == typeof(RepositoryXml<Course>));
```

Comment:
Why do you change it? It should be injected by IoC!!!

Reply:
Brought it back everywhere